OpenFeature/src/main/java/dev/openfeature/sdk/async/AsyncClient.kt

```diff
@@ -0,0 +1,32 @@
+package dev.openfeature.sdk.async
+
+import dev.openfeature.sdk.OpenFeatureClient
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+
+interface AsyncClient {
+    fun observeBooleanValue(key: String, default: Boolean): Flow<Boolean>
+    fun observeIntValue(key: String, default: Int): Flow<Int>
+    fun observeStringValue(key: String, default: String): Flow<String>
+}
+
+internal class AsyncClientImpl(
+    private val client: OpenFeatureClient
+) : AsyncClient {
+    private fun <T> observeEvents(callback: () -> T) = observeProviderReady()
```

Comment:
Can we have a "observeProviderState" that listens to any Provider State event?

Reply:
we have 2 functions one `observeProviderReady` and another one a suspending function `awaitProviderReady()`. added to the description of the PR.

Reply:
i removed the secondary way and now we have one way as helper functions to listen for providerReady 
```
CoroutineScope(Dispatchers.IO).launch {
        awaitProviderReady()
        // now provider is ready, read the properties
    }
```